03-pipefilter.md

```diff
@@ -16,277 +16,465 @@ Now that we know a few basic commands,
 ~~~
 ~~~ {.output}
-  9  methane.pdb
+   10 methanol.pdb
+   10 vinyl-chloride.pdb
+   12 ethane.pdb
+   13 cyclopropane.pdb
+   13 ethanol.pdb
+   14 glycol.pdb
+   15 lactic-acid.pdb
+   15 propane.pdb
+   16 cyclobutane.pdb
+ 1808 total
+   18 benzaldehyde.pdb
+   20 cubane.pdb
+   20 styrene.pdb
+   21 pentane.pdb
+   22 cinnamaldehyde.pdb
+   23 cyclohexanol.pdb
+   23 vanillin.pdb
+  248 lanoxin.pdb
+   24 ascorbic-acid.pdb
+   25 pyridoxal.pdb
+   25 tnt.pdb
+   30 aldrin.pdb
+   30 camphene.pdb
+   30 octane.pdb
+   33 citronellal.pdb
+   33 nerol.pdb
+   34 tyrian-purple.pdb
+   35 menthol.pdb
+   36 mint.pdb
+   41 mustard.pdb
+   42 thiamine.pdb
+   43 piperine.pdb
+   44 morphine.pdb
+   47 codeine.pdb
+   49 lactose.pdb
+   49 maltose.pdb
+   49 sucrose.pdb
+   51 norethindrone.pdb
+   51 strychnine.pdb
+   52 quinine.pdb
+   53 lsd.pdb
+   53 testosterone.pdb
+   54 ethylcyclohexane.pdb
+   54 tuberin.pdb
+   55 vitamin-a.pdb
+   78 cholesterol.pdb
+   79 heme.pdb
+    7 ammonia.pdb
+    9 methane.pdb
 ~~~
 
-The vertical bar between the two commands is called a **pipe**.
-It tells the shell that we want to use
-the output of the command on the left
-as the input to the command on the right.
-The computer might create a temporary file if it needs to,
-or copy data from one program to the other in memory,
-or something else entirely;
-we don't have to know or care.
-
-We can use another pipe to send the output of `wc` directly to `sort`,
-which then sends its output to `head`:
+Ops. Something went catastrophic wrong.
+By default, `sort` doesn't now that `1` is the first number,
+`2` is number after `1`, ..., `10` is the number after `9`
+and so on.
+We can request that `sort` interpret `1` as the first number,
+`2` as the number after `1`, ...
+by using the `-n` flag.
 
 ~~~ {.bash}
-$ wc -l *.pdb | sort -n | head -1
+$ wc -l *.pdb | sort
 ~~~
 ~~~ {.output}
-  9  methane.pdb
+    7 ammonia.pdb
+    9 methane.pdb
+   10 methanol.pdb
+   10 vinyl-chloride.pdb
+   12 ethane.pdb
+   13 cyclopropane.pdb
+   13 ethanol.pdb
+   14 glycol.pdb
+   15 lactic-acid.pdb
+   15 propane.pdb
+   16 cyclobutane.pdb
+   18 benzaldehyde.pdb
+   20 cubane.pdb
+   20 styrene.pdb
+   21 pentane.pdb
+   22 cinnamaldehyde.pdb
+   23 cyclohexanol.pdb
+   23 vanillin.pdb
+   24 ascorbic-acid.pdb
+   25 pyridoxal.pdb
+   25 tnt.pdb
+   30 aldrin.pdb
+   30 camphene.pdb
+   30 octane.pdb
+   33 citronellal.pdb
+   33 nerol.pdb
+   34 tyrian-purple.pdb
+   35 menthol.pdb
+   36 mint.pdb
+   41 mustard.pdb
+   42 thiamine.pdb
+   43 piperine.pdb
+   44 morphine.pdb
+   47 codeine.pdb
+   49 lactose.pdb
+   49 maltose.pdb
+   49 sucrose.pdb
+   51 norethindrone.pdb
+   51 strychnine.pdb
+   52 quinine.pdb
+   53 lsd.pdb
+   53 testosterone.pdb
+   54 ethylcyclohexane.pdb
+   54 tuberin.pdb
+   55 vitamin-a.pdb
+   78 cholesterol.pdb
+   79 heme.pdb
+  248 lanoxin.pdb
+ 1808 total
 ~~~
 
-This is exactly like a mathematician nesting functions like *log(3x)*
```

Comment:
I'd like to keep this paragraph as well - right before the Process callout, maybe?